pkg/tls/cert.go

```diff
@@ -25,13 +25,20 @@ type KeyPair struct {
 	KeyPEM  []byte
 }
 
-func NewSelfSignedCert(commonName string, hosts ...string) (KeyPair, error) {
+type CertType string
+
+const (
+	ServerCertType CertType = "server"
+	ClientCertType CertType = "client"
+)
+
+func NewSelfSignedCert(commonName string, certType CertType, hosts ...string) (KeyPair, error) {
```

Comment:
Just as a note (not a blocker for this PR): alternatively to `certType CertType` we could use `option` pattern (like we do in `ResourceStore` operations)